extensions/amp-story/1.0/page-advancement.js

```diff
@@ -721,10 +733,53 @@ class MediaBasedAdvancement extends AdvancementConfig {
     /** @private {?../../../src/video-interface.VideoInterface} */
     this.video_ = null;
 
-    if (element.ownerDocument.defaultView) {
-      /** @private @const {!./amp-story-store-service.AmpStoryStoreService} */
-      this.storeService_ =
-        getStoreService(element.ownerDocument.defaultView);
+    /** @private @const {!./amp-story-store-service.AmpStoryStoreService} */
+    this.storeService_ = getStoreService(win);
+
+    this.elements_.forEach(el => {
+      listen(el, VideoEvents.VISIBILITY, () => this.onVideoVisibilityChange_());
+    });
+  }
+
+  /**
+   * Returns the first displayed element, or null.
+   * @return {?Element}
+   * @private
+   */
+  getFirstDisplayedElement_() {
+    if (this.elements_.length === 1) {
+      return this.elements_[0];
+    }
+
+    for (let i = 0; i < this.elements_.length; i++) {
+      const element = this.elements_[i];
+      const resource = this.resources_.getResourceForElement(element);
+      if (element.getAttribute('layout') === Layout.NODISPLAY ||
```

Comment:
Why are we returning the first element with `layout=nodisplay`? It seems counterintuitive from the name and description of this method.